jetty-client/src/main/java/org/eclipse/jetty/client/util/AbstractRequestContent.java

```diff
@@ -0,0 +1,275 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under
+// the terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0
+//
+// This Source Code may also be made available under the following
+// Secondary Licenses when the conditions for such availability set
+// forth in the Eclipse Public License, v. 2.0 are satisfied:
+// the Apache License v2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.client.util;
+
+import java.nio.ByteBuffer;
+
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.util.BufferUtil;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.thread.AutoLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>Partial implementation of {@link Request.Content}.</p>
+ * <p>Manages a single subscription at a time (multiple simultaneous subscriptions are not allowed).</p>
+ */
+public abstract class AbstractRequestContent implements Request.Content
+{
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractRequestContent.class);
+
+    private final AutoLock lock = new AutoLock();
+    private final String contentType;
+    private Throwable failure;
```

Comment:
This is still not stateless.
Consider a ByteBufferContent that is shared between many requests to many servers... if just one of those request fails in the queue before a connection is made (perhaps it is an unknown host), then this failure will be set on the shared content here and will fail all the requests that share this content.

A reusable content must be stateless.  A non-reusable one can have state like failure if really needed.